src/security/Io.cc

```diff
@@ -83,10 +83,15 @@ Security::Handshake(Comm::Connection &transport, const ErrorCode topError, Fun i
         ; // fall through to handle the problem
     }
 
-    // now we know that we are dealing with a real problem; detail it
-    const ErrorDetail::Pointer errorDetail =
-        new ErrorDetail(topError, ioError, xerrno);
-
+    ErrorDetail::Pointer errorDetail;
+    if (const auto verifyErrorDetailRaw = SSL_get_ex_data(connection, ssl_ex_index_ssl_error_detail)) {
+        errorDetail = *static_cast<ErrorDetail::Pointer*>(verifyErrorDetailRaw);
```

Comment:
I do not think this code can or should know exactly what kind of detail ssl_ex_index_ssl_error_detail contains.

```suggestion
    if (const auto oldDetail = SSL_get_ex_data(connection, ssl_ex_index_ssl_error_detail)) {
        errorDetail = *static_cast<ErrorDetail::Pointer*>(oldDetail);
```

Reply:
commit c4de550fe3